### DISCUSSION TITLE
Option to start espanso in mode disabled

### DISCUSSION THREAD

**Discussion by Masacroso** [espanso/espanso#1192] [Feature requests and Ideas]
This is an enhancement/feature petition.

I would like to have an option to start espanso with the system but disabled instead of enabled to trigger substitutions. The point is that I use espanso in many of my sessions (almost all) to type latex in many different software and documents so for me is not an option to set the programs where I want that espanso trigger the replacements of strings (by example I use it also to type latex in web browsers but I need that most of the time espanso stay disabled). So every time I start the system I must tap twice the `Alt` key to disable espanso triggering, so the ideal solution for my workflow would be just start espanso but disabled, instead of enabled, so when I tap the `Alt` key twice then espanso enables the triggering. Reading the documentation I dont found a way to setup this behavior so I assume such option is not possible at this time, so I had written this petition.

**Comment by federico-terzi** [comment]
Hey @Masacroso,

What OS are you using? If you manage to execute this command after espanso starts you can achieve what you want:

```
espanso cmd disable
```

Cheers :)

**Comment by Masacroso** [comment]
> 
> 
> Hey @Masacroso,
> 
> What OS are you using? If you manage to execute this command after espanso starts you can achieve what you want:
> 
> ```
> espanso cmd disable
> ```
> 
> Cheers :)

Im using espanso in kubuntu 20.04. I know how to launch this command at the startup of the system but Im not sure how to launch it _after_ espanso is running. Maybe setting some delay in the execution will work, but I thought it could be a more elegant solution.

**Comment by federico-terzi** [comment]
Great, then you could do the following:
Edit the `~/.config/systemd/user/espanso.service` to add the following line below `ExecStart`

```
ExecStartPost={{path}} cmd disable
```

Make sure to replace `{{path}}` with the same espanso path found in `ExecStart`.

Maybe you will need to add a `sleep 1` in there as well.

This should solve the issue, for references: https://www.freedesktop.org/software/systemd/man/systemd.service.html#ExecStartPre=

Cheers :)

**Comment by St0rml** [comment]
Sorry to revive this closed issue, but my use case basically is the same as @Masacroso with the exception that I use espanso on Windows 10 Home as well as Kubuntu 20.04. While the update of the systemd service in Ubuntu can be done in the matter of seconds, to achieve this on Windows there is quite a few more hoops to jump through. I followed this tutorial: http://woshub.com/run-script-when-app-opens-closes/
This took me quite some time, especially because I use Windows 10 Home which does not provide the builtin Windows tools used in this tutorial, so I had to manually download/activate those with some sketchy batch scripts I found online.
It therefore might be a good idea to consider implementing the option to launch espanso in the disabled state? It would certainly help Windows users (I don't know about Mac users though)

Also I've been using espanso for quite a while now and I'm still loving it :) Great Job @federico-terzi !

**Comment by federico-terzi** [comment]
@St0rml Thank you for the feedback! I'll reopen the issue :)
